### PR TITLE
Mapped deprecated terms

### PR DESCRIPTION
**Pull request by joeflack4**
Previous PR name: Slurp: (i) re-run of outputs, (ii) deprecated analysis (needs generalization from OMIM into slurp)

## Updates
    Slurp
    - Update: slurp/omim.tsv and all artefact dependencies, following latest release addressing OMIM issues #98 (invalid refs OMIM:1 -> OMIM:14) and #84 (OMIM terms with 'None' as label).

    Deprecated terms in Mondo
    - Add: Script and make goals for this. Decoupled this from slurp/
    - Add: reports/omim_mapped_deprecated_terms.robot.template.tsv: A list of obsolete/deprecated OMIM terms that remain within Mondo.
    - Add: Docs

    Mapping progress report
    - Updates: Docs: Removed some unnecessary things / simplified some things

    Misc
    - Updates: Scripting: --sssom-map-path --> --mondo-mappings-path

## Reviewers
- Harshad & Vinícius: general comments
- Nicole and Sabrina to inspect outputs.
- Nico to inspect [`reports/omim_mapped_deprecated_terms.robot.template.tsv`](https://github.com/monarch-initiative/mondo-ingest/pull/108/commits/66e7654801be2767d0a9be2dd54ff527ee9f20eb#diff-b0d295ecd7d05fde88526ba07223a9bfbc08cc572d5ead04804bd3d49d2741b5) and related scripting.

### CODE REVIEW

**Comment by joeflack4** [comment]
@matentzn I believe this will close the loop on #98. If you wanna keep it open though, feel free to remove it from **Development** on the right of the page.

**Comment by joeflack4** [comment]
@sabrinatoro Ready for your review! If everything checks out, feel free to merge~

**Comment by matentzn** [comment]
Most of the new omim terms still need to be excluded..

**Comment by joeflack4** [comment]
Hmm, I'm not familiar with what is meant by new terms. And I'm not sure if this is something you are asking Sabrina to do in the exclusion tables, or if this is a bug in my script? (maybe you don't know either). Let me know if anything more is needed from me on this~

**Comment by matentzn** [comment]
Probably something @sabrinatoro needs to deal with, but we may need a more general solution for exclusions here. Lets wait for her feedback.

**Comment by sabrinatoro** [comment]
Do I understand correctly that the terms in slurp/omim.tsv _would be_ new terms in OMIM that _could be_ added into Mondo? 
Most (maybe all) of these should be excluded from Mondo because they are not true diseases (some are genes, some are traits). I couldn't find any of these OMIM ids in any of the lists I reviewed, so these omim id are probably "new". 
I think the way to move this forward is for me to review this new list, add the appropriate exclusion code, and add them to the excluded list. There is a possibility that a few of them are new that would need to be added into Mondo. I will know more after I have reviewed this list. 
There are less than 300 lines, I should be able to get to this "quickly".

**Comment by matentzn** [comment]
Let's try to exclude them all with one row, I.e by excluding SO:gene and all children, rather than 300 separate rows.

File: src/ontology/slurp/omim.tsv
Comment (nicolevasilevsky) [inline src/ontology/slurp/omim.tsv:18]: @joeflack4 could you pls find out where these types of terms come from?

Comment (joeflack4) [reply src/ontology/slurp/omim.tsv:18]: Just commenting that I've seen this. If I understand correctly, I believe this is one that we discussed on Mondo call today and that you will be opening a separate issue about this to assign to me.

In particular, the one case I remember us looking at involved something _baboon_ related. FYI I did a simple text search on "baboon" in all the OMIM files, and only a few rows came up. They were in:
- `mimTitles.txt`
- `genemap2.txt`

Comment (nicolevasilevsky) [reply src/ontology/slurp/omim.tsv:18]: yep, we discussed this. I can open a separate issue.

File: src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv
Comment (joeflack4) [inline src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv:1]: 91 mapped deprecated terms out of 1,345 deprecated OMIM terms

File: src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv
Comment (joeflack4) [inline src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv:3]: @matentzn I think that when you asked me to add `MONDO:equivalentObsolete` in this column, it is an `exclusion_reason`. But FYI, it doesn't look like we have that exact entry in [the exclusion list](https://mondo.readthedocs.io/en/latest/editors-guide/exclusion-reasons/) yet.

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:347]: @matentzn I made a section in the `makefile` called "Analysis" and added the goal there. I also put the Python scripting in `src/analysis/`.

There is one other script in `src/analysis`, but it's specific to exclusions, so I left that one under the "exclusions" section instead of here.

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:349]: @matentzn I wasn't sure if I should fully automate this, but I tried automating this part after finishing everything else:

`runoak -i omim.ttl obsoletes`

I tried running this in the `makefile` goal, but got a Python error: `PermissionError: [Errno 13] Permission denied: '/.data`. [stacktrace.txt](https://github.com/monarch-initiative/mondo-ingest/files/10117610/stacktrace.txt)

I know that the error is coming from OAK, but I don't know if this is something that ODK is somehow supposed to handle, or if this really is an OAK problem. What do you think?

File: src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv
Comment (matentzn) [reply src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv:3]: No no, its not an exclusion reason, made a fix. Will discuss details later.

Comment (joeflack4) [reply src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv:3]: Oh nooo, I just overwrote your commit; not sure what you changed. I'm so sorry.

Comment (joeflack4) [reply src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv:3]: Oh, I see your [suggestion](https://github.com/monarch-initiative/mondo-ingest/pull/108#discussion_r1035823464), `A oboInOwl:source`. I'll add that to the 2nd row, and change the header to `source` for now.

File: src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv
Comment (joeflack4) [reply src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv]: Implemented the suggestion.

File: src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv
Comment (matentzn) [reply src/ontology/reports/omim_mapped_deprecated_terms.robot.template.tsv:3]: Nice, thanks

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:349]: we can merge now. it runs outside of ODK. when pystow issue is fixed, we can run in ODK

**Comment by joeflack4** [comment]
@sabrinatoro I know we were going to discuss over the holidays but for this PR in particular, I think I was just asked to run the OMIM slurp and I figured you would want to look at the outputs. If you want to take a look at them, feel free. I will probably need to re-run the slurp anyway after this is merged: https://github.com/monarch-initiative/omim/pull/72

**Comment by joeflack4** [comment]
@matentzn This has been ready to merge for a while. I just went through again and verified evrything resolved.

**Review by matentzn — CHANGES_REQUESTED** [review]
I should have checked this earlier, sorry - this needs to be generalised to _all_ sources, the way the other slurps work!

**Comment by joeflack4** [comment]
@matentzn I'm going to deprioritize this until after the holidays. I thought this was originally an ad hoc analysis not even necessarily requiring scripting. But it seems useful for all ontologies. Here's the list of things I'll need to automate when I get back to this.

# TODO: This is ideally to be automated. There are some matching todo's above the makefile goal.
1. Makefile: Depend on goal: https://github.com/monarch-initiative/omim/releases/latest/downloads/omim.ttl
2. Run OAK
    2.a.1. Makefile: `runoak -i omim.ttl obsoletes > deprecated_terms.txt`
    2.a.2. Makefile?: deprecated_terms.txt: Need to remove " ! <label>" from each line
      - Example row: "https://omim.org/entry/100500 ! None"
    2.b.1. Chris suggested: `runoak -i omim.ttl term-metadata .all -O csv -p owl:deprecated`
    2.b.2. Some other transformations would be necessary after that too.
    2.c. @OAK: When https://github.com/INCATools/ontology-access-kit/issues/366 is merged, can do
      `runoak -i omim.ttl term-metadata .all -O csv -p owl:deprecated`
4. Python: Using `curies` and metadata/omim.yml, convert URIs to CURIEs

File: README.md
Comment (joeflack4) [inline README.md:16]: FYI: docs

File: docs/developer/workflows.md
Comment (joeflack4) [inline docs/developer/workflows.md:23]: FYI: more docs

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:464]: FYI: even more docs

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:372]: FYI: `make` goals

File: docs/reports/mapped_deprecated_doid.md
Comment (joeflack4) [inline docs/reports/mapped_deprecated_doid.md:2]: FYI: example results table for a given ontology

File: docs/reports/mapped_deprecated.md
Comment (joeflack4) [inline docs/reports/mapped_deprecated.md:1]: FYI: stats table

File: src/scripts/deprecated_in_mondo.py
Comment (joeflack4) [inline src/scripts/deprecated_in_mondo.py:34]: FYI: How the docs are generated

File: src/scripts/deprecated_in_mondo.py
Comment (joeflack4) [inline src/scripts/deprecated_in_mondo.py:64]: FYI: How the table artefacts are created

**Review by joeflack4 — COMMENTED** [review]
This PR reports on deprecated terms in a given ontology that have existing xrefs in Mondo.

@matentzn This has been generalized, and I have added outputs in Markdown format as well. This was previously just an ad hoc analysis for OMIM. This also leverages the `mapping_status` TSVs I created during the mapping progress report work which greatly simplified what needed to be done here. There's some redundancy w/ slurp too which will likely lead to optimizations. Will discuss during our meeting.

@sabrinatoro I think you will find this useful! Just look at the [home page of `mondo-ingest` ](https://github.com/monarch-initiative/mondo-ingest)and you'll see a link to these tables! 

@hrshdhgd @souzadevinicius Just tagging you guys for a general review. Let me know if you have any comments / curiosities / suggestions.

**Comment by matentzn** [comment]
This is _amazing_ work. Can you also get your docs pages to deploy nicely here: https://monarch-initiative.github.io/mondo-ingest

I think all that is needed is adding them into mkdocs.yml

**Comment by joeflack4** [comment]
Oh yeah I forgot that part. Will do!

**Pull request merged**
